### PR TITLE
New Website and Logo

### LINKED ISSUE #400: ProbNum Landing Page
Alternatively we should also be able to host the `readthedocs` site at `probnum.org`.

### PR DESCRIPTION
**Pull request by JonathanWenger**
# In a Nutshell
New website and design for ProbNum: https://probnum--433.org.readthedocs.build/en/433/

# Detailed Changes
- introduces a new logo for ProbNum
- adds a landing page for ProbNum
- surpasses http://probabilistic-numerics.org by integrating the meetings and literature page
- modernizes the documentation look and feel (by switching to `pydata-sphinx-theme` in combination with custom CSS)

This closes #400.

# Minor Changes and Fixes
- Swapped out markdown parser for the Sphinx recommended option [MyST](https://myst-parser.readthedocs.io/en/latest/)
- Documentation fixes for `probsolve_ivp` and `LinearOperator.trace()`
- Citations are now included via new doc dependency `sphinxcontrib-bibtex`

# ToDos after Review
- [x] change links to resources from `JonathanWenger/probnum` to `probabilistic-numerics/probnum`
- [ ] Setup http://probnum.org as an alternative to https://probnum.readthedocs.io via ReadTheDocs

### CODE REVIEW

**Review by mmahsereci — COMMENTED** [review]
This looks really great, Jonathan! I made some minor comments. I guess the 2 main points are:

- [x] it's hard to parse the literature section (see comment there).
- [ ] I cannot read the code snippets well at all (contrast too low). Could we use a slightly lighter font perhaps.

File: docs/source/_templates/index.html [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        </div>
+        <div class="highlight-bash notranslate">
+            <div class="highlight highlight-max-width">
+                <pre><span></span>@misc{probnum2021,
+    author    = {Wenger, Jonathan and Kr{\" a}mer, Nicholas and Bosch, Nathanael
+                 and Pf{\" o}rtner, Marvin and Schmidt, Jonathan and Effenberger, Nina
+                 and Mahsereci, Maren and Zenn, Johannes and Gl{\" a}{\ss}le, Thomas
+                 and Gessner, Alexandra and Karvonen, Toni and Hennig, Philipp
+                 and {ProbNum Contributors}},
+    title     = {ProbNum: Probabilistic Numerics in Python},
```
[the remark this answers is not in the document]
Comment (JonathanWenger) [reply docs/source/_templates/index.html:108]: maybe it would be best to keep this out of this PR to not hold up the design on the authorship discussion?

[the remark this answers is not in the document]
Comment (mmahsereci) [reply docs/source/_templates/index.html:108]: sure

File: docs/source/research/literature.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+## General and Foundational
+The following papers are often cited as early works on the
+idea of uncertainty over the result of deterministic computations. Some entries have a "notes" field providing further information about the relevance of the cited work, or pointers to specific results therein.
+
+<!-- {% bibliography --file general %} -->
+```{bibliography} bibliography/general.bib
+---
+all:
+list: bullet
+---
```
[the remark this answers is not in the document]
Comment (JonathanWenger) [reply docs/source/research/literature.md]: Using `sphinxcontrib-bibtex` this was not super easy to solve. I settled on changing the citation style to `unsrtalpha` to keep the reference order from the file (currently by date) and avoid label duplications across different sub-bibliographies.

**Review by philipphennig — COMMENTED** [review]
This looks great! 

--- 

## Literature
On _a small number_ of the entries in the original page we have a "notes" section, which we introduced to add additional information on why this particular reference relates to PN. Would it be possible to add those somehow? 

Just out of interest: Is there an easy way to add our publications efficiently to the literature list? I'm asking because, of course, many of our more recent papers aren't on the list. I'm not even sure it would be a good idea to add them specifically, we could perhaps also just add a link to our publication list (and that of the groups of Tim Sullivan, Chris Oates, etc.), so that we're not flooding the list with just our stuff. But it's a bit odd that we have some papers on there, and not others. 

## Meetings and events: 

This is great. I'm still a bit unsure about loosing the explicit style of the old pages of individual meetings (it's breaking a bit with the idea of them being archival). But I don't think it's a crucial issue.

## The Rest

* Is there a difference between *Quickstart* and *Tutorials*? Or is the former a subset of the latter?
* In the long run, we could make it a deliverable that each PhD student leaves a tutorial on the methods they developed in their PhD on the page (and such tutorials could leave the page when they're deprecated by later work). But this just as a passing idea.
* The bottom of the page has a "(C) Probnum Team" entry, but it's not a link to any list. Is there a list of contributors somewhere on the page? I understand that we want to wait with the citation, but it would still be good to say a bit about who is primarily contributing, and if it's just to give people an idea who to contact.

**Comment by mmahsereci** [comment]
I can perhaps say sth about the tutorial page: It's being restructured currently so the content on that page will change in the next couple of weeks. Roughly, there will be 1 quickstart for the ProbNum library in general (this is a common thing to do), and 1 quickstart per subpackage (BQ, lin solver, ODE, etc). The rest will also look a bit different, and the namings will be more descriptive. 

I like the idea a lot for each PhD student to add a tutorial alongside new functionality!

File: docs/source/research/literature.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmahsereci) [reply docs/source/research/literature.md]: looks much better now

**Comment by JonathanWenger** [comment]
> On _a small number_ of the entries in the original page we have a "notes" section, which we introduced to add additional information on why this particular reference relates to PN. Would it be possible to add those somehow?

Unfortunately this is very difficult (impossible?) with the way we import citations currently. The notes are preserved in the bibtex file, but unless we change how we import and display the citations, there is no good way of showing the notes field.

> Just out of interest: Is there an easy way to add our publications efficiently to the literature list? I'm asking because, of course, many of our more recent papers aren't on the list. I'm not even sure it would be a good idea to add them specifically, we could perhaps also just add a link to our publication list (and that of the groups of Tim Sullivan, Chris Oates, etc.), so that we're not flooding the list with just our stuff. But it's a bit odd that we have some papers on there, and not others.

I added a "Note", which links to creating a pull request with a new `bibtex` entry. I couldn't come up with a process that is fair to everyone. At the end we are working with the library a lot, so we will be curating the list. However, anyone can add an entry via PR.

> ## Meetings and events:
> 
> This is great. I'm still a bit unsure about loosing the explicit style of the old pages of individual meetings (it's breaking a bit with the idea of them being archival). But I don't think it's a crucial issue.
> ## The Rest
> 
>     * Is there a difference between _Quickstart_ and _Tutorials_? Or is the former a subset of the latter?
> 
>     * In the long run, we could make it a deliverable that each PhD student leaves a tutorial on the methods they developed in their PhD on the page (and such tutorials could leave the page when they're deprecated by later work). But this just as a passing idea.

Highly in favor of this idea. We are already going for this a little bit, but making this an explicit deliverable is great for the library and excellent exposure for the corresponding paper(s).

>     * The bottom of the page has a "(C) Probnum Team" entry, but it's not a link to any list. Is there a list of contributors somewhere on the page? I understand that we want to wait with the citation, but it would still be good to say a bit about who is primarily contributing, and if it's just to give people an idea who to contact.

I linked to the contributor list.

**Comment by JonathanWenger** [comment]
@philipphennig I also added the Bundesministerium BW and the ERC logo on the landing page as supporters. Is there any other sponsor / funding organization that should be listed?

**Review by mmahsereci — APPROVED** [review]
Looks good to me. After @philipphennig commented on the logos.
